doc/source/rayserve/overview.rst

```diff
@@ -0,0 +1,197 @@
+
+To follow along, you'll need to make the necessary imports.
+
+.. code-block:: python
+
+  from ray import serve
+  serve.init() # initializes serve and Ray
+
+
+Endpoints
+~~~~~~~~~
+
+Endpoints allow you to name the "entity" that you'll be exposing, 
+the HTTP path that your application will expose. 
+Endpoint are "logical" and decoupled from the business logic or 
+model that you'll be serving. To create one, we'll simply specify the name, route, and methods.
+
+.. code-block:: python
+
+  serve.create_endpoint("my_endpoint", "/my_path", methods=["GET", "POST"])
+
+Backends
+~~~~~~~~
+
+Backends are the logical structures for your business logic or models and 
+how you specify what should happen when an endpoint is queried.
+To define a backend, first you must define the "handler" or the business logic you'd like to respond with. 
+The input to this request will be a `Flask Request object <https://flask.palletsprojects.com/en/1.1.x/api/?highlight=request#flask.Request>`_.
+Once you define the function (or class) that will handle a request. 
+You'd use a function when your response is stateless and a class when you
+might need to maintain some state (like a model). 
+For both functions and classes (that take as input Flask Requests), you'll need to 
+define them as backends to RayServe.
+
+It's important to note that RayServe places these backends in individual workers, which are replicas of the model.
+
+.. code-block:: python
+  
+  def handle_request(flask_request):
+    return "hello world"
+
+  class RequestHandler:
+    def __init__(self):
+        self.msg = "hello, world!"
+
+    def __call__(self, flask_request):
+        return self.msg
+
+  serve.create_backend(handle_request, "my_endpoint_backend")
+  serve.create_backend(RequestHandler, "my_endpoint_backend_class")
+
+Lastly, we need to link the particular backend to the server endpoint. 
+To do that we'll use the ``link`` capability.
+A link is essentially a load-balancer and allow you to define queuing policies 
+for how you would like backends to be served via an endpoint.
+For instance, you can route 50% of traffic to Model A and 50% of traffic to Model B.
+
+.. code-block:: python
+
+  serve.link("my_endpoint_backend", "my_endpoint")
+
+Once we've done that,we can now query our endpoint with something like requests 
+(or an external service).
+
+.. code-block:: python
+  
+  import requests
+  print(requests.get("http://127.0.0.1:8000/-/routes", timeout=0.5).text)
+
+Configuring Backends
+~~~~~~~~~~~~~~~~~~~~
+
+There are a number of things you'll likely want to do with your serving application including
+scaling out, splitting traffic, or batching input for better response performance. To do all of this,
+you will create a ``BackendConfig``, a configuration object that you'll use to set 
+the properties of a particular backend. That will allow you to do things like scale out,
+split traffic, and batch requests for better performance.
```

Comment:
last sentence is redundant with first sentence?

Reply:
100%

Reply:
fixed